test/module/irohad/ordering/on_demand_os_client_grpc_test.cpp

```diff
@@ -37,28 +38,31 @@ struct OnDemandOsClientGrpcTest : public ::testing::Test {
   OnDemandOsClientGrpc::TimepointType timepoint;
   std::chrono::milliseconds timeout{1};
   std::shared_ptr<OnDemandOsClientGrpc> client;
-  Round round{1, 2};
+  consensus::Round round{1, 2};
 };
 
 /**
  * @given client
- * @when onTransactions is called
+ * @when onBatches is called
  * @then data is correctly serialized and sent
  */
-TEST_F(OnDemandOsClientGrpcTest, onTransactions) {
-  proto::TransactionsRequest request;
+TEST_F(OnDemandOsClientGrpcTest, onBatches) {
+  proto::BatchesRequest request;
   auto r = std::make_unique<
       MockClientAsyncResponseReader<google::protobuf::Empty>>();
-  EXPECT_CALL(*stub, AsyncSendTransactionsRaw(_, _, _))
+  EXPECT_CALL(*stub, AsyncSendBatchesRaw(_, _, _))
       .WillOnce(DoAll(SaveArg<1>(&request), Return(r.get())));
 
   OdOsNotification::CollectionType collection;
   auto creator = "test";
   protocol::Transaction tx;
   tx.mutable_payload()->mutable_reduced_payload()->set_creator_account_id(
       creator);
-  collection.push_back(std::make_unique<shared_model::proto::Transaction>(tx));
-  client->onTransactions(round, std::move(collection));
+  collection.push_back(
+      std::make_unique<shared_model::interface::TransactionBatchImpl>(
+          shared_model::interface::types::SharedTxsCollectionType{
```

Comment:
Can it guess the type, without `shared_model::interface::types::SharedTxsCollectionType` ? Just {...}

Reply:
`error: no matching function for call to 'make_unique(<brace-enclosed initializer list>)'`